cli/internal/cmd/upgradecheck.go

```diff
@@ -81,6 +81,7 @@ func runUpgradeCheck(cmd *cobra.Command, args []string) error {
 			flags:          flags,
 			cliVersion:     compatibility.EnsurePrefixV(constants.VersionInfo()),
 			log:            log,
+			verFetcher:     fetcher.NewFetcher(),
```

Comment:
```suggestion
			versionsapi:     fetcher.NewFetcher(),
```

Maybe? `versionCollector.versionsapi.FetchVersionList(ctx, list)` reads a bit more clear to me.